### PR TITLE
Log gradient norm in grpo_fast

### PR DESCRIPTION
**Pull request by mnoukhov**
## Summary
- log DeepSpeed gradient norm in grpo_fast after optimizer updates
- surface non-finite grad norms in the logged metric instead of filtering them out
- document that the collected grad norms may include nan/inf values

## Testing
- make style
- make quality

### CODE REVIEW

File: open_instruct/grpo_fast.py [hunk 1]
```diff
@@ -725,6 +726,9 @@ def step(self):
                     self.model.backward(loss)
                     if is_accumulation_boundary:
                         self.model.step()
+                        grad_norm = self.model.get_global_grad_norm()
```
Comment (finbarrtimbers) [inline open_instruct/grpo_fast.py]: When would this be None?

Comment (mnoukhov) [reply open_instruct/grpo_fast.py]: I also thought that was weird and codex initially convinced me it was possible in theory

but I now realize that as long as we're using a decently normal optimizer, it won't be None so I will remove the check

File: open_instruct/grpo_fast.py [hunk 2]
```diff
@@ -745,6 +747,8 @@ def step(self):
             batch_metrics = batch_data["metrics"]
             with torch.no_grad():
                 self._compute_loss_metrics(loss_stats_B, total_valid_tokens)
+                if grad_norms:
```
Comment (finbarrtimbers) [inline open_instruct/grpo_fast.py]: Don't think this is needed?

Comment (mnoukhov) [reply open_instruct/grpo_fast.py]: you're right, removing
